src/plots/cartesian/axes.js

```diff
@@ -2636,6 +2638,35 @@ axes.drawOne = function(gd, ax, opts) {
     return Lib.syncOrAsync(seq);
 };
 
+function keepSelectedAutoMargin(automargin, push) {
+    if(typeof automargin === 'boolean') return push;
+
+    var keepMargin = [];
+    var mapping = {
+        width: ['x', 'r', 'l'],
+        height: ['y', 't', 'b'],
+        right: ['r'],
+        left: ['l'],
+        top: ['t'],
+        bottom: ['b']
+    };
+
+    Object.keys(mapping).forEach(function(key) {
+        if(automargin.includes(key)) {
```

Comment:
😢 still need to support IE11 so [can't use .includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes#browser_compatibility)

Reply:
fixed